src/parser.js

```diff
@@ -386,24 +437,114 @@ export default class Parser extends Stream {
             },
             'skip'() {
               this.manifest.skip = entry.attributes;
+
+              if (!entry.attributes.hasOwnProperty('SKIPPED-SEGMENTS')) {
+                this.trigger('warn', {
+                  message: '#EXT-X-SKIP lacks required attribute SKIPPED-SEGMENTS'
+                });
+              }
             },
             'part'() {
               this.manifest.parts = this.manifest.parts || [];
               this.manifest.parts.push(entry.attributes);
+              const missingAttributes = [];
+
+              ['URI', 'DURATION'].forEach(function(k) {
+                if (!entry.attributes[k]) {
+                  missingAttributes.push(k);
+                }
+              });
+
+              if (missingAttributes.length) {
+                const index = this.manifest.parts.length - 1;
+
+                this.trigger('warn', {
+                  message: `#EXT-X-PART #${index} lacks required attribute(s) ${missingAttributes.join(', ')}`
+                });
+              }
+
+              if (this.manifest.renditionReports) {
+                this.manifest.renditionReports.forEach((r, i) => {
+                  if (!r.hasOwnProperty('LAST-PART')) {
+                    this.trigger('warn', {
+                      message: `#EXT-X-RENDITION-REPORT #${i} lacks required attribute(s) LAST-PART`
+                    });
+                  }
+                });
+              }
             },
             'server-control'() {
-              this.manifest.serverControl = entry.attributes;
+              const attrs = entry.attributes;
+
+              this.manifest.serverControl = attrs;
+              if (!attrs.hasOwnProperty('CAN-BLOCK-RELOAD')) {
+                this.manifest.serverControl['CAN-BLOCK-RELOAD'] = false;
+                this.trigger('info', {
+                  message: '#EXT-X-SERVER-CONTROL defaulting CAN-BLOCK-RELOAD to false'
+                });
+              }
+              setHoldBack.call(this, this.manifest);
+
+              if (attrs.hasOwnProperty('CAN-SKIP-DATERANGES') && !attrs.hasOwnProperty('CAN-SKIP-UNTIL')) {
```

Comment:
are these props only set if their values are YES?

Reply:
I think this should be a boolean check here, as we only expect CAN-SKIP-UNTIL if `CAN-SKIP-DATERANGES` is `YES` (which will be true here already.